### PR TITLE
added "how-to" guide on IBM MQ server TLS setup

### PR DESCRIPTION
**Pull request by rekies**
## Describe changes

After several customers asking similar questions we decided to start a new section for "how-to guides for API-M. Idea is to add knowledge from the field to share it as part of the product docs as one-stop shop.

I created a new sub-category for "how-to guides" and started with **IBM MQ server integration and adoption of TLS**. More how-to's shall follow.

## Checklist for contributors

Before submitting this PR, please make sure:

* [x] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [x] You have signed the [Axway CLA](https://cla.axway.com/)
* [x] You have verified the technical accuracy of your change
* [x] You have verified that your change does not expose any sensitive information (passwords, keys, etc.)
* [x] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)

## Checklist for approvers

_This section is to be filled out by project maintainers only._

Before approving this PR, please make sure it:

* [x] Does not have conflicts with the base branch
* [x] Is clear and complete
* [x] Is believed to be accurate (technical accuracy to be checked with an SME for PRs originating from outside R&D)
* [x] Follows [Axway style](https://techweb.axway.com/confluence/display/RIE/Style+guide)
* [x] Follows [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)
* [x] Follows [best practices](https://axway-open-docs.netlify.com/docs/contribution_guidelines/bestpracticedevdoc/)
* [x] Does not contain typos, grammatical errors, etc.
* [x] Does not expose any sensitive information
* [x] Passes the Markdown linter rules (automated via CI check)
* [x] Does not contain broken links
* [x] Is discoverable and navigable

## Checklist for mergers

_This section is to be filled out by project maintainers only._

Before merging this PR, please make sure:

* [x] You have applied a size label (unless this work is tracked in the ReadMeFirst JIRA project)
* [x] You have applied a production label if this needs a manual push to Zoomin
* [x] You have added it to the correct GitHub project
* [x] You have added it to the correct Sprint milestone
* [x] You have appended a `fixes` line to this description if this PR fixes a specific GitHub issue
* [x] You have added a new collection to Netlify CMS config (static/admin/config.js) if required (new doc sections)
* [x] You have updated the Zoomin classification file if required (new AMPC topics)

### CODE REVIEW

**Comment by rekies** [comment]
Andreas and Alex, Chris has advocated a lot and now I'm following his approach. Hope that works for you.

**Comment by alexearnshaw** [comment]
> Andreas and Alex, Chris has advocated a lot and now I'm following his approach. Hope that works for you.

@rekies Thanks so much, this looks good, I will review today and get back to you if I have any questions.

**Comment by cwiechmann** [comment]
@alexearnshaw, I think the phrase: "How to" is a special term and people are very used to it. Therefore I would propose to call that section: "How to" instead of how-to.

**Comment by alexearnshaw** [comment]
> @alexearnshaw, I think the phrase: "How to" is a special term and people are very used to it. Therefore I would propose to call that section: "How to" instead of how-to.

@cwiechmann I did check this as I wasn't familiar with using this term and we don't have a style rule for it. Most dictionaries say that how-to is the correct usage whether as an adjective or noun (e.g. https://www.merriam-webster.com/dictionary/how-to)

Is your concern around people searching for "how to" as I believe most search engines will ignore the hyphen in any case, so users should find this section regardless of searching for "how-to" or "how to"?

**Comment by cwiechmann** [comment]
@alexearnshaw, I mainly added this proposal, because it took me some time to find this new How-To chapter in the list and I knew it was there. @rekies finally pointed me on the correct entry.

Perhaps because I was "scanning" the list for "How to ...." and not the other way around and I think this is what other people will also do. 

What do you think about: 
"How To's for API-Management" ? << My favorit
"How Tos for API-Management" ?

I know, this isn't perfect English but it is used quite often like so.

**Comment by alexearnshaw** [comment]
> @alexearnshaw, I mainly added this proposal, because it took me some time to find this new How-To chapter in the list and I knew it was there. @rekies finally pointed me on the correct entry.
> 
> Perhaps because I was "scanning" the list for "How to ...." and not the other way around and I think this is what other people will also do.
> 
> What do you think about:
> "How To's for API-Management" ? << My favorit
> "How Tos for API-Management" ?
> 
> I know, this isn't perfect English but it is used quite often like so.

Ah ok, I can change the ordering so that how-tos appears first as you suggest, but the correct form would be "How-tos for API management". This terms does look quite strange and I can't find many other doc sites using it so I'm wondering now is there a better alternative? Maybe "Best practices for API management"? Or do you think that would be even worse for people scanning?

**Comment by rekies** [comment]
Yes, The "How-to" at the beginning helps to find it. - I have seen this at IBM too. (did a quick Google search for: _**ibm "how to" guides**_)

File: content/en/docs/apim_howto_guides/apigw_mqclient_tlssetup.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+## Integration with IBM MQ
+
+One of the core components of AMPLIFY API Management is the API Gateway. API Gateway ships with support for several third-party systems for easier integrations. API Gateway is acting as a client requesting services from an external provider. One of those built-in features is support for the messaging system [IBM MQ](https://www.ibm.com/products/mq). IBM MQ is a widely used enterprise message communication service that often powers complex and mission critical services for large enterprises.
+
+API Gateway is built on Java and uses Java message service (JMS) for messaging. Connecting to JMS servers requires some general configuration and special settings:
+
+* Connect to JMS server configuration - see [Configure messaging services](/docs/apim_policydev/apigw_poldev/general_messaging/)
+* To listen on queues or topics for messages to process, configure a JMS service provider (listener) - see [Configure API Gateway instances and listeners](/docs/apim_policydev/apigw_gw_instances/) on section *Configure Messaging System connections*
```
Comment (alexearnshaw) [inline content/en/docs/apim_howto_guides/apigw_mqclient_tlssetup.md:16]: @rekies I'm trying to find a live link to replace this "on section *Configure Messaging System connections*" but I can't find anything in the linked instances and listeners topic. Wondering if this link would be better to reference in this line https://axway-open-docs.netlify.app/docs/apim_policydev/apigw_poldev/general_messaging/#configure-a-jms-service ? Can you confirm?

Comment (rekies) [reply content/en/docs/apim_howto_guides/apigw_mqclient_tlssetup.md:16]: Ah, yes. Sorry, forgot to mention this! There is an error in the docs right now. This section is available in the old docs (up to 7.6.2) but is missing here. All listener are stated, but the JMS listener has not mad its way to the new docs yet. 
The original description here [Configure a JMS session](https://docs.axway.com/bundle/APIGateway_762_PolicyDevGuide_allOS_en_HTML5/page/Content/PolicyDevTopics/general_messaging_session.htm) is a bit wired. Guess we need to rework this.

The point is, the *message listener* config description is missing in OpenDocs. The list of listeners is provided here [Configure API Gateway instances and listeners](https://axway-open-docs.netlify.app/docs/apim_policydev/apigw_gw_instances/). - And Policy Studio shows "Messaging System" in addition. - Given this logic, I expected the JMS listener configuration to be in that list.

The JMS capabilities were described differently on a special section before: [Messaging](https://docs.axway.com/bundle/APIGateway_762_PolicyDevGuide_allOS_en_HTML5/page/Content/PolicyDevTopics/part_messaging.htm). I guess that's the reason it is missing..

Comment (rekies) [reply content/en/docs/apim_howto_guides/apigw_mqclient_tlssetup.md:16]: Shall I assist with re-arranging / re-writing this part?

Comment (alexearnshaw) [reply content/en/docs/apim_howto_guides/apigw_mqclient_tlssetup.md:16]: I might have misunderstood what's missing in open docs - but I think the message listener config is in open docs here https://axway-open-docs.netlify.app/docs/apim_policydev/apigw_poldev/general_messaging/#configure-a-jms-session ? Can you confirm if that is the right topic and if so I will change the link on this line to point to that instead.

And yes please, if that JMS session topic needs some rework, do feel free to update it in another pull request :)

Comment (rekies) [reply content/en/docs/apim_howto_guides/apigw_mqclient_tlssetup.md:16]: I have re-read this chapter again. - Yes, that's the right one to refer to from my how-to for MQ TLS.

What I'd like to add is an short description for the listener configuration for messaging services. All other listener from http to file poller are descibed at one place. Only messaging is not mentioned there. Having a short notice and over there and than referring to the detailed messaging description would be the right thing to do.  I just was looking at the wrong place. I guess other will have the same issue.  ;-)

**Comment by rekies** [comment]
@alexearnshaw, I have seen you removed the notes on last visited the links. I added them, because I thought they are useful in case the link target will change and we will not notice it. At university I was forced to correctly quote my sources - including this last visited statement for links.. So, for our docs it's sufficient to set normal links and have the source control for the time reference?

**Comment by alexearnshaw** [comment]
> @alexearnshaw, I have seen you removed the notes on last visited the links. I added them, because I thought they are useful in case the link target will change and we will not notice it. At university I was forced to correctly quote my sources - including this last visited statement for links.. So, for our docs it's sufficient to set normal links and have the source control for the time reference?

@rekies Yes I removed them as although it can be useful, it's not common practice on doc sites. It can also give the impression to users that our docs are not maintained for example if those dates are still there in 5 years time. To handle the issue with the links changing, we don't have this just yet, but we do intend to automate link checking in all our docs as part of our CI flow, which will catch any dead links as soon as they occur. 

There's a couple of other changes I wanted to explain also before I merge this:

- I removed the subfolder that contained the IBM MQ topic itself. I didn't think there was any value in having this right now as it just adds an extra click for a user to navigate to the topic. However, if at some point additional topics will be added for IBM MQ it might make sense to have a subfolder (but we can do that then).
- I changed the link format slightly for internal links as only this part of the link is needed (`/docs/apim_policydev/apigw_poldev/general_messaging/`), that is, `index.html` is not needed at the end and in fact might not work when to page is published on docs.axway.com. (The easiest way to get the links in the right format is to copy them from the Open docs site https://axway-open-docs.netlify.app/)

I think all my other changes relate to either Axway style rules or are general improvements to improve the clarity and conciseness of the text, but please check my changes and comment if I've changed something I shouldn't have or removed something important.

**Comment by andreamussap** [comment]
can I just add my 2 cents to this already long conversation?? 

_"How-tos for API management". This terms does look quite strange_ --> it really does. Why don't you do the same as [IBM, How-to Guide Overview](https://www.ibm.com/support/knowledgecenter/SSMPHH_9.5.0/com.ibm.guardium95.doc/how_to/topics/how-to_guide_overview.html): This help book presents a series of how-to topics....

This is the documentation for API Management so you don't need to specify it in the title of the menu?? Couldn't it be only 'How-to guides', or then, 'How-to guides for API management'?

**Comment by cwiechmann** [comment]
> Couldn't it be only 'How-to guides', or then, 'How-to guides for API management'?

I was thinking at the moment everything we do, goes into one book (the Open-Docs) no matter if API-Management, Streams, Central, ... and maybe more to come. 
Therefore I was thinking having API-Management as part of the title is better. 

Without that constraint I would be really happy to go with: 'How-to guides'. And anyway, I like both options you suggested.

**Comment by alexearnshaw** [comment]
@andreamussap @cwiechmann @rekies I've changed to "How-to guides for API management" for now. When we do move API Central and other content out to other repos, we can delete the API management part, but I think for now best to leave it.

**Comment by alexearnshaw** [comment]
### Thank you
We really appreciate your contribution and thank you for helping us to improve this documentation!
 
### Where to find the updated documentation
An Axway technical writer has reviewed and merged your PR, which is now available on the following sites:
Netlify (staging): https://axway-open-docs.netlify.com/docs/
Zoomin (production): https://docs.axway.com/bundle/axway-open-docs/
 
### Send us your feedback
If you have a moment, please complete the following 2 minute survey to give us your feedback on this process and help us make it even better:
https://www.surveymonkey.com/r/PJ5HQFD

**Pull request merged**